geodiff/src/drivers/sqlitedriver.cpp

```diff
@@ -49,20 +49,29 @@ class Sqlite3DbMutexLocker
 class Sqlite3SavepointTransaction
 {
   public:
-    explicit Sqlite3SavepointTransaction( std::shared_ptr<Sqlite3Db> db )
-      : mDb( db )
+    explicit Sqlite3SavepointTransaction( std::shared_ptr<Sqlite3Db> db, const Context *context )
+      : mDb( db ), mContext( context )
     {
       if ( sqlite3_exec( mDb.get()->get(), "SAVEPOINT changeset_apply", 0, 0, 0 ) != SQLITE_OK )
-        throw GeoDiffException( "Unable to start savepoint" );
+      {
+        std::string errMsg = sqliteErrorMessage( mDb.get()->get(), "Sqlite3SavepointTransaction" );
+        throw GeoDiffException( "Unable to start savepoint: " + errMsg );
```

Comment:
You can create function  `throwSqliteError(db, logger, function/details)` for these similar to `logSqliteError`